lib/beaker-puppet/install_utils/foss_utils.rb

```diff
@@ -1448,6 +1450,7 @@ def install_puppetserver_on(host, opts = {})
 
           if opts[:version] == 'latest' || opts[:nightlies]
             release_stream += '-nightly' unless release_stream.end_with? "-nightly"
+            opts[:version] = nil
```

Comment:
Does this need to be `nil`, or is the point to make it "not set"? Would `opts.delete(:version)` make sense?

Reply:
I think it does need to be nil. Otherwise, `install_package` below will try to pass in `latest` as the version number and it'll fail. If "not set" gets `opts[:version` to return `nil`, then that would also work.

Reply:
```
irb(main):001:0> a = { :version => nil }
=> {:version=>nil}
irb(main):002:0> a[:version]
=> nil
irb(main):003:0> a.delete(:version)
=> nil
irb(main):004:0> a[:version]
=> nil
```

There's no difference between the two as far as what `opts[:version]` would return, it's more about how merging the hash with other hashes, and how `opts.has_key?(:version)` would behave. There's also the slight difference in that `opts.delete(:version)` reads a bit more like "we don't care about the version whatever it was", but there aren't a ton of differences in how the code would behave.